src/Microsoft.OData.Client/DataServiceContext.cs

```diff
@@ -3935,13 +4018,89 @@ private void DeleteObjectInternal(object entity, bool failIfInAddedState)
                     Util.IncludeLinkState(state),
                     "bad state transition to deleted");
 
+                if (dependsOnIds != null)
+                {
+                    //In DataServiceContext.UpdateObject, we pass entity Id(s) as dependsOnIds.
+                    //We will extract ChangeOrder from the entity descriptor based on the Id.
+                    //This is because we pass the ChangeOrder as Content-ID to the batch writer.
+                    //We will pass the Content-ID(s) as dependsOnIds to the batch writer.
+                    List<string> dependsOnIdsAsChangeOrders;
+                    HashSet<string> dependsOnChangeSetIds;
+
+                    GetDependsOnChangeOrdersAndChangeSetIds(
+                        dependsOnIds,
+                        this.EntityTracker.Entities,
+                        this.Model,
+                        out dependsOnIdsAsChangeOrders,
+                        out dependsOnChangeSetIds);
+
+                    resource.DependsOnIds = dependsOnIdsAsChangeOrders;
+                    resource.DependsOnChangeSetIds = dependsOnChangeSetIds.ToList();

```

Comment:
[Question] Once this value is set? do we loop through it multiple times or could we even refactor it out so it can take either hashset or an IEnumerable? Is this property exposed to the world at the moment?

Reply:
We pass the value as it is when calling `ODataBatchWriter`.`CreateOperationsRequestMessage` https://github.com/OData/odata.net/blob/4e263ca4b94d01f2d105d8282c62ec272cfcd1bb/src/Microsoft.OData.Client/ODataRequestMessageWrapper.cs#L126

The non-async `CreateOperationsRequestMessage` accepts `IEnumerable<string>` https://github.com/OData/odata.net/blob/4e263ca4b94d01f2d105d8282c62ec272cfcd1bb/src/Microsoft.OData.Core/Batch/ODataBatchWriter.cs#L294-L295
However the async method accepts `IList<string>` https://github.com/OData/odata.net/blob/4e263ca4b94d01f2d105d8282c62ec272cfcd1bb/src/Microsoft.OData.Core/Batch/ODataBatchWriter.cs#L339-L340
That's why I decided to stick with `List<string>`